api/resourcedeleter.go

```diff
@@ -0,0 +1,123 @@
+package api
+
+import (
+	"k8s.io/client-go/kubernetes"
+	k8smeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"fmt"
+	redisclient "github.com/spotahome/redis-operator/client/k8s/clientset/versioned/typed/redisfailover/v1alpha2"
+	"k8s.io/client-go/rest"
+)
+
+
+func deleteK8sResouces(namespace string, deployName string, k8sClient kubernetes.Interface) error {
```

Comment:
It would be clearer if all or none of the delete operations were behind a functions.

Reply:
We could then also easier give different HTTP-errors for each of the resources we couldn't delete.